pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -915,47 +879,45 @@ func (p *processorImpl) EnqueuePiggybackedAdmittedAtLeader(
 	var admitted [raftpb.NumPriorities]uint64
 	copy(admitted[:], state.Admitted)
 
-	p.mu.leader.pendingAdmittedMu.Lock()
-	defer p.mu.leader.pendingAdmittedMu.Unlock()
+	p.leader.pendingAdmittedMu.Lock()
+	defer p.leader.pendingAdmittedMu.Unlock()
 	// Invariant: if updates == nil, we are not the leader or already stepping
 	// down. Ignore the updates in this case.
-	if p.mu.leader.pendingAdmittedMu.updates == nil {
+	if p.leader.pendingAdmittedMu.updates == nil {
 		return
 	}
 	// Merge in the received admitted vector. We are only interested in the
 	// highest admitted marks. Zero value always merges in favour of the new one.
-	p.mu.leader.pendingAdmittedMu.updates[from] =
-		p.mu.leader.pendingAdmittedMu.updates[from].Merge(
+	p.leader.pendingAdmittedMu.updates[from] =
+		p.leader.pendingAdmittedMu.updates[from].Merge(
 			rac2.AdmittedVector{Term: state.Term, Admitted: admitted})
 }
 
 // ProcessPiggybackedAdmittedAtLeaderRaftMuLocked implements Processor.
 func (p *processorImpl) ProcessPiggybackedAdmittedAtLeaderRaftMuLocked(ctx context.Context) bool {
 	p.opts.Replica.RaftMuAssertHeld()
-	p.mu.Lock()
-	defer p.mu.Unlock()
-	if p.mu.destroyed {
+	if p.destroyed {
 		return false
 	}
 	var updates map[roachpb.ReplicaID]rac2.AdmittedVector
 	// Swap the updates map with the empty scratch. This is an optimization to
 	// minimize the time we hold the pendingAdmittedMu lock.
 	func() {
-		p.mu.leader.pendingAdmittedMu.Lock()
-		defer p.mu.leader.pendingAdmittedMu.Unlock()
-		if updates = p.mu.leader.pendingAdmittedMu.updates; len(updates) != 0 {
-			p.mu.leader.pendingAdmittedMu.updates = p.mu.leader.scratch
-			p.mu.leader.scratch = updates
+		p.leader.pendingAdmittedMu.Lock()
+		defer p.leader.pendingAdmittedMu.Unlock()
+		if updates = p.leader.pendingAdmittedMu.updates; len(updates) != 0 {
+			p.leader.pendingAdmittedMu.updates = p.leader.scratch
+			p.leader.scratch = updates
 		}
 	}()
 	if len(updates) == 0 {
 		return false
 	}
 	for replicaID, state := range updates {
-		p.mu.leader.rc.AdmitRaftMuLocked(ctx, replicaID, state)
+		p.leader.rc.AdmitRaftMuLocked(ctx, replicaID, state)
 	}
 	// Clear the scratch from the updates that we have just handled.
-	clear(p.mu.leader.scratch)
+	clear(p.leader.scratch)
 	return true
```

Comment:
@sumeerbhola btw, I think we no longer need this bool return value. Since the updates are pushed to `AdmitRaftMuLocked` immediately, do we still need to schedule Ready? Previously we had to because we updated the admitted vectors, but then `RangeController` had to fetch and apply them.

Reply:
Closing this comment. Let's discuss this one in #130585.